File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/presentation/dialog/draft_list/DraftListDialog.java
package com.bits.bee.bpmc.presentation.dialog.draft_list;

import java.lang.System;

/**
 * Created by aldi on 24/06/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000N\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0010\u000b\n\u0002\b\u0004\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0002\b\u0004\n\u0002\u0010\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\b\u0007\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001B)\u0012\"\b\u0002\u0010\u0003\u001a\u001c\u0012\u0004\u0012\u00020\u0005\u0012\u0006\u0012\u0004\u0018\u00010\u0006\u0012\u0004\u0012\u00020\u0007\u0012\u0004\u0012\u00020\u00020\u0004\u00a2\u0006\u0002\u0010\bJ\b\u0010\u0018\u001a\u00020\u0019H\u0016J\u001a\u0010\u001a\u001a\u00020\u00192\u0006\u0010\u001b\u001a\u00020\u001c2\b\u0010\u001d\u001a\u0004\u0018\u00010\u001eH\u0016J\u0010\u0010\u001f\u001a\u00020\u00192\u0006\u0010 \u001a\u00020\u0007H\u0002J\b\u0010!\u001a\u00020\u0019H\u0016J\b\u0010\"\u001a\u00020\u0019H\u0016R.\u0010\u0003\u001a\u001c\u0012\u0004\u0012\u00020\u0005\u0012\u0006\u0012\u0004\u0018\u00010\u0006\u0012\u0004\u0012\u00020\u0007\u0012\u0004\u0012\u00020\u00020\u0004X\u0096\u0004\u00a2\u0006\b\n\u0000\u001a\u0004\b\t\u0010\nR\u000e\u0010\u000b\u001a\u00020\fX\u0082.\u00a2\u0006\u0002\n\u0000R\u001b\u0010\r\u001a\u00020\u000e8BX\u0082\u0084\u0002\u00a2\u0006\f\n\u0004\b\u0011\u0010\u0012\u001a\u0004\b\u000f\u0010\u0010R\u001b\u0010\u0013\u001a\u00020\u00148BX\u0082\u0084\u0002\u00a2\u0006\f\n\u0004\b\u0017\u0010\u0012\u001a\u0004\b\u0015\u0010\u0016\u00a8\u0006#"}, d2 = {"Lcom/bits/bee/bpmc/presentation/dialog/draft_list/DraftListDialog;", "Lcom/bits/bee/bpmc/presentation/base/BaseBottomSheetDialogFragment;", "Lcom/bits/bee/bpmc/databinding/DialogDraftListBinding;", "bindingInflater", "Lkotlin/Function3;", "Landroid/view/LayoutInflater;", "Landroid/view/ViewGroup;", "", "(Lkotlin/jvm/functions/Function3;)V", "getBindingInflater", "()Lkotlin/jvm/functions/Function3;", "draftAdapter", "Lcom/bits/bee/bpmc/presentation/dialog/draft_list/DraftDialogAdapter;", "mViewModel", "Lcom/bits/bee/bpmc/presentation/ui/pos/MainViewModel;", "getMViewModel", "()Lcom/bits/bee/bpmc/presentation/ui/pos/MainViewModel;", "mViewModel$delegate", "Lkotlin/Lazy;", "viewModel", "Lcom/bits/bee/bpmc/presentation/dialog/draft_list/DraftListViewModel;", "getViewModel", "()Lcom/bits/bee/bpmc/presentation/dialog/draft_list/DraftListViewModel;", "viewModel$delegate", "initComponents", "", "onViewCreated", "view", "Landroid/view/View;", "savedInstanceState", "Landroid/os/Bundle;", "setVisibilityEmpty", "isVisible", "subscribeListeners", "subscribeObservers", "app_debug"})
@dagger.hilt.android.AndroidEntryPoint()
public final class DraftListDialog extends com.bits.bee.bpmc.presentation.base.BaseBottomSheetDialogFragment<com.bits.bee.bpmc.databinding.DialogDraftListBinding> {
    @org.jetbrains.annotations.NotNull()
    private final kotlin.jvm.functions.Function3<android.view.LayoutInflater, android.view.ViewGroup, java.lang.Boolean, com.bits.bee.bpmc.databinding.DialogDraftListBinding> bindingInflater = null;
    private final kotlin.Lazy viewModel$delegate = null;
    private final kotlin.Lazy mViewModel$delegate = null;
    private com.bits.bee.bpmc.presentation.dialog.draft_list.DraftDialogAdapter draftAdapter;
    
    public DraftListDialog() {
        super();
    }
    
    public DraftListDialog(@org.jetbrains.annotations.NotNull()
    kotlin.jvm.functions.Function3<? super android.view.LayoutInflater, ? super android.view.ViewGroup, ? super java.lang.Boolean, com.bits.bee.bpmc.databinding.DialogDraftListBinding> bindingInflater) {
        super();
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public kotlin.jvm.functions.Function3<android.view.LayoutInflater, android.view.ViewGroup, java.lang.Boolean, com.bits.bee.bpmc.databinding.DialogDraftListBinding> getBindingInflater() {
        return null;
    }
    
    private final com.bits.bee.bpmc.presentation.dialog.draft_list.DraftListViewModel getViewModel() {
        return null;
    }
    
    private final com.bits.bee.bpmc.presentation.ui.pos.MainViewModel getMViewModel() {
        return null;
    }
    
    @java.lang.Override()
    public void onViewCreated(@org.jetbrains.annotations.NotNull()
    android.view.View view, @org.jetbrains.annotations.Nullable()
    android.os.Bundle savedInstanceState) {
    }
    
    @java.lang.Override()
    public void initComponents() {
    }
    
    @java.lang.Override()
    public void subscribeListeners() {
    }
    
    @java.lang.Override()
    public void subscribeObservers() {
    }
    
    private final void setVisibilityEmpty(boolean isVisible) {
    }
}

File: app/src/main/java/com/bits/bee/bpmc/presentation/dialog/draft_list/DraftListDialog.kt
package com.bits.bee.bpmc.presentation.dialog.draft_list

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.view.isVisible
import androidx.fragment.app.activityViewModels
import androidx.fragment.app.viewModels
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.bits.bee.bpmc.R
import com.bits.bee.bpmc.databinding.DialogDraftListBinding
import com.bits.bee.bpmc.presentation.base.BaseBottomSheetDialogFragment
import com.bits.bee.bpmc.presentation.ui.pos.MainViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.launch

/**
 * Created by aldi on 24/06/22.
 */
@AndroidEntryPoint
class DraftListDialog(
    override val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> DialogDraftListBinding = DialogDraftListBinding::inflate
) : BaseBottomSheetDialogFragment<DialogDraftListBinding>() {

    private val viewModel : DraftListViewModel by viewModels()
    private val mViewModel : MainViewModel by activityViewModels()

    private lateinit var draftAdapter : DraftDialogAdapter

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel.getDraftList()
    }

    override fun initComponents() {
        binding.apply {
            draftAdapter = DraftDialogAdapter(
                onItemClick = {
                    viewModel.onItemClick(it)
                }
            )
            rvList.apply {
                adapter = draftAdapter
                layoutManager = LinearLayoutManager(requireActivity())
            }
        }
    }

    override fun subscribeListeners() {
        binding.apply {
            btnTutup.setOnClickListener {
                dismiss()
            }
            imageView2.setOnClickListener {
                dismiss()
            }
            btnLihatSelengkapnya.setOnClickListener {
                val action = DraftListDialogDirections.actionDraftListDialogToDraftFragment()
                findNavController().navigate(action)
            }
        }
    }

    override fun subscribeObservers() {
        viewLifecycleOwner.lifecycleScope.launch {
                viewModel.event.collect {
                    when(it){
                        is DraftListViewModel.UIEvent.RequestDraft -> {
                            mViewModel.loadDraft(it.sale)
                            findNavController().popBackStack(R.id.posFragment, false)
                        }
                    }
                }

        }
        viewLifecycleOwner.lifecycleScope.launch {
            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED){
                viewModel.viewStates().collect {
                    it?.let {
                        draftAdapter.submitList(it.saleList)
                        setVisibilityEmpty(it.saleList.isEmpty())
                    }
                }
            }
        }
    }

    private fun setVisibilityEmpty(isVisible : Boolean) {
        binding.apply {
            rvList.isVisible = !isVisible
            btnLihatSelengkapnya.isVisible = !isVisible
            group2.isVisible = isVisible
        }
    }

}